polaris-core/src/main/java/org/apache/polaris/core/entity/PolarisEvent.java

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ */
+
+package org.apache.polaris.core.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+
+public class PolarisEvent {
+  public static final String EMPTY_MAP_STRING = "{}";
+
+  // to serialize/deserialize properties
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  // catalog id
+  private String catalogId;
+
+  // event id
+  private String id;
+
+  // id of the request that generated this event
+  private String requestId;
+
+  // event type that was fired
+  private String eventType;
+
+  // timestamp in epoch milliseconds of when this event was emitted
+  private long timestampMs;
+
+  // polaris principal who took this action
+  private String principalName;
+
+  // Enum that states the type of resource was being operated on
+  private ResourceType resourceType;
+
+  // Which resource was operated on
+  private String resourceIdentifier;
+
+  // Additional parameters that were not earlier recorded
+  private String additionalProperties;
+
+  public String getCatalogId() {
+    return catalogId;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public String getRequestId() {
+    return requestId;
+  }
+
+  public String getEventType() {
+    return eventType;
+  }
+
+  public long getTimestampMs() {
+    return timestampMs;
+  }
+
+  public String getPrincipalName() {
+    return principalName;
+  }
+
+  public ResourceType getResourceType() {
+    return resourceType;
+  }
+
+  public String getResourceIdentifier() {
+    return resourceIdentifier;
+  }
+
+  public String getAdditionalProperties() {
+    return additionalProperties != null ? additionalProperties : EMPTY_MAP_STRING;
+  }
+
+  public PolarisEvent(
+      String catalogId,
+      String id,
+      String requestId,
+      String eventType,
+      long timestampMs,
+      String actor,
+      ResourceType resourceType,
+      String resourceIdentifier) {
+    this.catalogId = catalogId;
+    this.id = id;
+    this.requestId = requestId;
+    this.eventType = eventType;
+    this.timestampMs = timestampMs;
+    this.principalName = actor;
+    this.resourceType = resourceType;
+    this.resourceIdentifier = resourceIdentifier;
+  }
+
+  @JsonIgnore
+  public void setAdditionalProperties(Map<String, String> properties) {
+    try {
+      this.additionalProperties = properties == null ? null : MAPPER.writeValueAsString(properties);
+    } catch (JsonProcessingException ex) {
+      throw new IllegalStateException(
+          String.format("Failed to serialize json. properties %s", properties), ex);
+    }
+  }
+
+  public void setAdditionalProperties(String additionalProperties) {
+    this.additionalProperties = additionalProperties;
+  }
+
+  public enum ResourceType {
+    CATALOG,
+    NAMESPACE,
+    TABLE,
+    VIEW
+  }
```

Comment:
What's the overall plan to add other types in the future?
How can the various kinds of tables be differentiated?